connexion/middleware/security.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import pathlib
+import typing as t
+from collections import defaultdict
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from connexion.apis.abstract import AbstractSpecAPI
+from connexion.http_facts import METHODS
+from connexion.lifecycle import MiddlewareRequest
+from connexion.middleware import AppMiddleware
+from connexion.middleware.routing import CONNEXION_CONTEXT
+from connexion.security import SecurityHandlerFactory
+
+logger = logging.getLogger("connexion.middleware.security")
+
+
+class SecurityMiddleware(AppMiddleware):
+    """Middleware to check if operation is accessible on scope."""
+
+    def __init__(self, app: ASGIApp) -> None:
+        self.app = app
+        self.apis = {}
+        self.operations = {}
+
+    def add_api(self, specification: t.Union[pathlib.Path, str, dict], **kwargs) -> None:
+        api = SecurityAPI(specification, **kwargs)
+        self.apis[api.base_path] = api
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send):
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+
+        try:
+            connexion_context = scope['extensions'][CONNEXION_CONTEXT]
+        except KeyError:
+            raise  # TODO
+
+        if api_base_path := connexion_context.get('api_base_path'):
+            api = self.apis[api_base_path]
+            operation_id = connexion_context.get('operation_id')
+            try:
+                operation = api.operations[operation_id]
+            except KeyError:
+                pass
+            else:
+                request = MiddlewareRequest(scope)
+                await operation(request)
+
+        await self.app(scope, receive, send)
+
+
+class SecurityAPI(AbstractSpecAPI):
+
+    def __init__(
+            self,
+            specification: t.Union[pathlib.Path, str, dict],
+            auth_all_paths: bool = False,
+            *args,
+            **kwargs
+    ):
+        super().__init__(specification, *args, **kwargs)
+        self.security_handler_factory = SecurityHandlerFactory('context')
```

Comment:
Hardcoded `pass_context_arg_name` here. 
Would propose to get rid of the parameter and choose a fixed name, but to do so in a follow-up PR.